src/Components/TaskScheduler.php

```diff
@@ -17,24 +19,95 @@ class TaskScheduler
+            }
+
             app(Schedule::class)
                 ->command('df:request \'' . $data . '\' ' . $commandOptions)
                 ->cron('*/' . $task->frequency . ' * * * *')
                 ->withoutOverlapping()
-//                ->appendOutputTo(storage_path() . '/scheduled-tasks-responses.log')
-                ->onFailure(function () use ($task) {
+                ->sendOutputTo($logFilePath)
+                ->onSuccess(function () use ($task, $logFilePath) {
+                    if (TaskLog::whereTaskId($task->id)->exists()) {
+                        TaskLog::whereTaskId($task->id)->first()->delete();
+                    }
+                    if (file_exists($logFilePath)) unlink($logFilePath);
+                })
+                ->onFailure(function () use ($task, $logFilePath) {
+                    try {
+                        $taskId = $task->id;
+                        $errorContent = file_exists($logFilePath) ? file_get_contents($logFilePath) : '';
+                        $statusCode = self::getStatusCode($logFilePath);
+                        $taskLog = new TaskLog(['task_id' => $taskId, 'status_code' => $statusCode, 'content' => $errorContent]);
+
+                        if (!TaskLog::whereTaskId($taskId)->exists()) {
+                            $taskLog->save();
+                        } else {
+                            $taskLog = TaskLog::whereTaskId($taskId)->first();
+                            $isSameError = Str::before($taskLog->content, "\n") === Str::before($errorContent, "\n");
+                            if ($taskLog->task_id === $taskId && $taskLog->status_code === $statusCode && $isSameError) {
+                                if (file_exists($logFilePath)) unlink($logFilePath);
+                                return;
+                            }
+                            $taskLog->status_code = $statusCode;
+                            $taskLog->content = $errorContent;
+                            $taskLog->save();
+                        }
+                        if (file_exists($logFilePath)) unlink($logFilePath);
+                    } catch (\Exception $e) {
+                        \Log::error('Could not store failed scheduler task log: ' . $e->getMessage());
+                    }
                 });
+
+            // close and remove log file after scheduling
+            if (file_exists($logFilePath)) fclose($logFile);
         }
     }
+
+    /**
+     * Get status code from error log
+     *
+     * @param string $fileName
+     * @return float|int|string
+     */
+    private static function getStatusCode($fileName)
+    {
+        $lines = file($fileName);
+        foreach ($lines as $lineNumber => $line) {
+            if (strpos($line, 'statusCode') !== false) {
+                return abs((int)filter_var($line, FILTER_SANITIZE_NUMBER_INT));
+            }
+        }
+        return '500';
```

Comment:
In line 98, we return `int`, but here we return 'string'

Reply:
made it int